ext/UnitfulExt.jl

```diff
@@ -280,13 +278,25 @@ Plots.process_limits(lims::Tuple{S,T}, axis) where {S<:Quantity,T<:Quantity} =
 
 function _ustrip(u, x)
     u isa MixedUnits && return ustrip(uconvert(u, x))
-    return ustrip(u, x)
+    ustrip(u, x)
 end
 
 function _unit(x)
-    t = eltype(x)
-    t <: LogScaled && return logunit(t)
-    return unit(x)
+    (t = eltype(x)) <: LogScaled && return logunit(t)
+    unit(x)
+end
+
+Plots.pgfx_sanitize_string(s::UnitfulString) = begin
+    tex_str = if (str_unit = string(s.unit)) == s.content
+        ""
+    else
+        Plots.pgfx_sanitize_string(s.content)
+    end
+    tex_unit = replace(Plots.wrap_power_label(str_unit), ' ' => '~')
+    if !Plots.pgfx_is_inline_math(tex_unit)
+        tex_unit = "\\($tex_unit\\)"  # force inline math mode
+    end
+    isempty(tex_str) ? tex_unit : tex_str * ' ' * tex_unit
```

Comment:
shoudn't this return a ` UnitfulString`?

Reply:
I think the issue https://github.com/JuliaPlots/Plots.jl/issues/4722 showed that this wouldn't work.